### PR TITLE
Upgrade to React Leaflet V4

### PR DESCRIPTION
**Pull request by JoshGx2000**
Upgrading to React Leaflet V4. Many changes, like we've discussed.

Everything should work as before except the leaflet-draw stuff. I've tried using leaflet-draw directly over react-leaflet-draw, but it's really not a good substitute.

I did tidy up a lot of my code, by expect you'll find stuff I've missed.

### CODE REVIEW

File: frontend/src/cljs/imas_seamap/interop/react.cljs [hunk 1]
```diff
@@ -6,7 +6,7 @@
             ["react" :as react]
             ["react-transition-group" :refer [TransitionGroup CSSTransition]]
             ["react-sizeme" :refer [SizeMe]]
-            ["react-leaflet-sidebarv2" :refer [Sidebar Tab]]))
+            #_["react-leaflet-sidebarv2" :refer [Sidebar Tab]]))
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/interop/react.cljs:9]: Yeah you can kill that!

File: frontend/src/cljs/imas_seamap/map/events.cljs [hunk 2]
```diff
@@ -454,24 +455,40 @@
...
+    (.on leaflet-map "mousemove"          #(re-frame/dispatch [:ui/mouse-pos {:x (-> % .-containerPoint .-x) :y (-> % .-containerPoint .-y)}]))
+    (.on leaflet-map "mouseout"           #(re-frame/dispatch [:ui/mouse-pos nil]))
+    (.on leaflet-map "easyPrint-start"    #(re-frame/dispatch [:ui/show-loading "Preparing Image..."]))
+    (.on leaflet-map "easyPrint-finished" #(re-frame/dispatch [:ui/hide-loading]))
+
+    (assoc-in db [:map :leaflet-map] leaflet-map)))
+
+(defn update-map-view [{{:keys [leaflet-map] old-zoom :zoom old-center :center} :map} [_ {:keys [zoom center bounds instant?]}]] 
+  (when leaflet-map
+    (if instant?
```
Comment (markhepburn) [inline frontend/src/cljs/imas_seamap/map/events.cljs:474]: I was goign to chase this up -- what's the purpose of "instant?"?  It looks rather hackish at first glance.

File: frontend/yarn.lock [hunk 3]
```diff
@@ -251,6 +251,11 @@
   resolved "https://registry.yarnpkg.com/@babel/helper-plugin-utils/-/helper-plugin-utils-7.17.12.tgz#86c2347da5acbf5583ba0a10aed4c9bf9da9cf96"
   integrity sha512-JDkf04mqtN3y4iAbO1hv9U2ARpPyPL1zqyWs/2WG1pgSq9llHFjStX5jdxb84himgJm+8Ng+x0oiWF/nw/XQKA==
 
+"@babel/helper-plugin-utils@^7.18.6":
```
Comment (markhepburn) [inline frontend/yarn.lock:254]: We have lockfiles for both yarn and npm I think; probably worth standardising on one or the other (it'll be the one the github action uses, I guess)

**Review by markhepburn — APPROVED** [review]

File: frontend/src/cljs/imas_seamap/map/events.cljs (hunk 2 above)
Comment (JoshGx2000) [reply frontend/src/cljs/imas_seamap/map/events.cljs:474]: Ah, it's a difference of whether we want the map to fly to that point vs just instantly moving there.

**Pull request merged**